make/image.mk

```diff
@@ -0,0 +1,110 @@
+# This is a wrapper to build and push docker image
+#
+# All make targets related to docker image are defined in this file.
+
+# Set MAKE_IN_DOCKER=1 as an environment variable to run `make` inside
+# a Docker container with preinstalled tools.
+
+DOCKER_BUILDER_IMAGE ?= envoyproxy/gateway-dev-builder
+DOCKER_BUILDER_TAG ?= latest
+DOCKER_BUILD_CMD ?= DOCKER_BUILDKIT=1 docker build
+DOCKER_RUN_CMD ?= docker run \
+		  --rm \
+		  -t \
+		  -v /var/run/docker.sock:/var/run/docker.sock \
+		  -v ${PWD}:/workspace
+
+%:
+ifeq ($(MAKE_IN_DOCKER), 1)
```

Comment:
this needs to be in the top level make file, else it doesn't work

Reply:
Done